### PR TITLE
Reverted the singleton & .of(context) implementation

### PR DESCRIPTION
**Pull request by vanlooverenkoen**
Reverting back to .of(context) and removed the singleton

In the last 2 projects we started. (Only 2 we started) we had the same issues when reloading the translations. 
The reason is that we need the context so the build method can be called again.

Documentation on why the context is needed for rebuilding can be found here. -> 
https://docs.flutter.dev/development/accessibility-and-localization/internationalization#loading-and-retrieving-localized-values

### CODE REVIEW

File: example/lib/viewmodel/locale/locale_viewmodel.dart [hunk 1]
```diff
@@ -19,10 +19,13 @@ class LocaleViewModel with ChangeNotifier {
...
-    await _localizationInstance.load(
-      locale: locale,
-      localizationOverrides: customLocalizationOverrides,
-    );
+    if (locale != null) {
+      localeDelegate = LocalizationDelegate(
+        newLocale: locale,
+        localizationOverrides: customLocalizationOverrides,
+      );
+      notifyListeners();
```
Comment (ikbendewilliam) [inline example/lib/viewmodel/locale/locale_viewmodel.dart:27]: this notify is redundant, see line 29

File: example/pubspec.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ flutter:
     - assets/locale/
 
 locale_gen:
-  languages: ['en','nl', 'zh-Hans-CN', 'fi-FI']
+  languages: [ 'en','nl', 'zh-Hans-CN', 'fi-FI' ]
```
Comment (ikbendewilliam) [inline example/pubspec.yaml:28]: spaces are a bit weird 😅

**Review by ikbendewilliam — APPROVED** [review]

File: example/pubspec.yaml (hunk 2 above)
Comment (vanlooverenkoen) [reply example/pubspec.yaml:28]: This is the default spacing 🤷

**Comment by vanlooverenkoen** [comment]
Fixing the build & setting up github actions as well

**Review by NicolaVerbeeck — APPROVED** [review]

**Pull request merged**